### PR TITLE
Seasonal-, theater-, weather-, and day/night-based temperature and pressure

### PR DESCRIPTION
**Pull request by magwo**
See some supporting data at the end of this comment.

# Tips from ATC guy
> 30.12 typical summer high pressure (nordics)
> 29.68 lower bound typical summer rain (nordics)
> conclusion: add/subtract 0.22 for clear vs rain
> 27.91 crazy low, seen
> 31.30 crazy high, seen
> Thunderstorm keeps pressure high
> conclusion: add 0.22 for thunderstorm
> Pressure doesn't change much during night
> Seasonal variations are smaller close to the equator

# Test runs

The interpolation logging is about summer, winter, resulting-value interpolation.
```

Caucasus
<game.weather.Cloudy object at 0x000001950FFD4640> 1995-06-01 TimeOfDay.Night
interpolated pressure between 30.02 29.72 29.969180327868852
interpolated temperature between 22.5 3.0 19.19672131147541
is night
Temperature is now 16.19672131147541
After weather type adjustment, temperature is 16.19672131147541
After weather type adjustment, pressure is 29.969180327868852
Randomizing and clamping...
Final conditions AtmosphericConditions(qnh_inches_mercury=29.926219391865057, temperature_celsius=16)

Caucasus
<game.weather.Raining object at 0x000001950FFD47F0> 1995-06-02 TimeOfDay.Dawn
interpolated pressure between 30.02 29.72 29.970819672131146
interpolated temperature between 22.5 3.0 19.30327868852459
Temperature is now 19.30327868852459
After weather type adjustment, temperature is 16.30327868852459
After weather type adjustment, pressure is 29.750819672131147
Randomizing and clamping...
Final conditions AtmosphericConditions(qnh_inches_mercury=29.643524585341304, temperature_celsius=16)

Caucasus
<game.weather.Cloudy object at 0x0000022C52DB0160> 1995-01-01 TimeOfDay.Day
interpolated pressure between 30.02 29.72 29.721639344262293
interpolated temperature between 22.5 3.0 3.1065573770491817
is day
Temperature is now 6.106557377049182
After weather type adjustment, temperature is 6.106557377049182
After weather type adjustment, pressure is 29.721639344262293
Randomizing and clamping...
Final conditions AtmosphericConditions(qnh_inches_mercury=29.81610288497605, temperature_celsius=6)

Caucasus
<game.weather.ClearSkies object at 0x0000022C488101C0> 1995-01-01 TimeOfDay.Dusk
interpolated pressure between 30.02 29.72 29.721639344262293
interpolated temperature between 22.5 3.0 3.1065573770491817
Temperature is now 3.1065573770491817
After weather type adjustment, temperature is 6.106557377049182
After weather type adjustment, pressure is 29.941639344262292
Randomizing and clamping...
Final conditions AtmosphericConditions(qnh_inches_mercury=29.94006951684297, temperature_celsius=3)


Marianas
<game.weather.Raining object at 0x000002204076B610> 1995-06-01 TimeOfDay.Day
interpolated pressure between 30.02 29.82 29.986120218579234
interpolated temperature between 28.0 27.0 27.830601092896174
is day
Temperature is now 28.330601092896174
After weather type adjustment, temperature is 25.330601092896174
After weather type adjustment, pressure is 29.766120218579236
Randomizing and clamping...
Final conditions AtmosphericConditions(qnh_inches_mercury=29.83005343317575, temperature_celsius=28)

Marianas
<game.weather.ClearSkies object at 0x000002204AA38580> 1995-06-01 TimeOfDay.Dusk
interpolated pressure between 30.02 29.82 29.986120218579234
interpolated temperature between 28.0 27.0 27.830601092896174
Temperature is now 27.830601092896174
After weather type adjustment, temperature is 30.830601092896174
After weather type adjustment, pressure is 30.206120218579233
Randomizing and clamping...
Final conditions AtmosphericConditions(qnh_inches_mercury=30.195798131957048, temperature_celsius=29)
```

# Data from my smart friend Urist
```

Map:  CA
HGPT noon temperature jan-dec    :
[ 9.9 11.9 15.  19.9 24.7 28.  30.5 30.8 27.4 22.3 16.5 11.5]
HGPT midnight temperature jan-dec:
[ 5.3  6.9  9.5 13.7 18.1 21.8 24.4 24.5 20.9 15.9 10.5  6.4]

Map:  SY
HGPT noon temperature jan-dec    :
[17.5 18.7 22.  26.5 31.1 35.2 37.8 38.6 36.3 31.1 24.9 19.9]
HGPT midnight temperature jan-dec:
[10.3 11.3 14.1 17.7 21.5 25.3 27.7 28.  25.7 21.  15.7 11.9]

Map:  PG
HGPT noon temperature jan-dec    :
[22.4 23.3 26.5 31.1 34.3 35.9 36.4 36.1 34.8 32.4 28.4 24.4]
HGPT midnight temperature jan-dec:
[20.5 21.4 24.5 29.1 32.5 34.1 34.7 34.  32.4 30.  26.1 22.3]

Map:  NTR
HGPT noon temperature jan-dec    :
[17.2 19.9 24.6 30.  35.3 40.8 43.8 42.9 37.9 30.  22.  17.8]
HGPT midnight temperature jan-dec:
[12.1 14.6 18.4 23.1 28.1 33.6 37.1 36.3 31.1 23.3 15.8 12.1]

Map:  MA
HGPT noon temperature jan-dec    :
[27.4 27.2 27.3 27.7 28.2 28.4 28.1 27.9 28.  28.1 28.2 28. ]
HGPT midnight temperature jan-dec:
[27.3 27.1 27.2 27.6 28.  28.2 27.9 27.7 27.8 27.9 28.  27.8]
```

### CODE REVIEW

File: game/theater/conflicttheater.py [hunk 1]
```diff
@@ -497,6 +497,17 @@ class ReferencePoint:
     image_coordinates: Point
 
 
+@dataclass
```
Comment (DanAlbert) [inline game/theater/conflicttheater.py]: `frozen=True`

File: game/theater/conflicttheater.py [hunk 2]
```diff
@@ -507,6 +518,7 @@ class ConflictTheater:
     land_poly = None  # type: Polygon
     """
     daytime_map: Dict[str, Tuple[int, int]]
+    seasonal_conditions: SeasonalConditions
```
Comment (DanAlbert) [inline game/theater/conflicttheater.py]: If you make this an abstract `@property` it won't affect save compat.

File: game/weather.py [hunk 3]
```diff
@@ -71,15 +72,73 @@ class Fog:
...
+        print("After weather type adjustment, pressure is", pressure)
+        print("Randomizing and clamping...")
+        conditions = AtmosphericConditions(
+            qnh_inches_mercury=self.random_pressure(pressure),
+            temperature_celsius=self.random_temperature(temperature),
+        )
+        print("Final conditions", conditions)
+        return conditions
+
+    def get_pressure_adjustment(self) -> float:
```
Comment (DanAlbert) [inline game/weather.py]: `@property`

File: game/weather.py [hunk 4]
```diff
@@ -71,15 +72,73 @@ class Fog:
...
+        raise NotImplementedError
+
+    def get_temperature_adjustment(self) -> float:
```
Comment (DanAlbert) [inline game/weather.py]: Ditto

File: game/weather.py [hunk 5]
```diff
@@ -259,7 +330,13 @@ def generate_start_time(
...
 
     @classmethod
-    def generate_weather(cls) -> Weather:
+    def generate_weather(
+        cls,
+        seasonal_conditions: SeasonalConditions,
+        day: datetime.date,
+        time_of_day: TimeOfDay,
+    ) -> Weather:
+        # Here we could hook in seasonal weights
```
Comment (magwo) [inline game/weather.py]: This is where the magic happens @DanAlbert

**Pull request merged**